File: urbanairship-core/src/main/java/com/urbanairship/app/SimpleActivityListener.java
/* Copyright Airship and Contributors */

package com.urbanairship.app;

import android.app.Activity;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * A convenience class to extend when you only want to listen for a subset
 * of of activity events.
 */
public class SimpleActivityListener implements ActivityListener {

    @Override
    public void onActivityPaused(@NonNull Activity activity) {
    }

    @Override
    public void onActivityCreated(@NonNull Activity activity, @Nullable Bundle bundle) {
    }

    @Override
    public void onActivityStarted(@NonNull Activity activity) {
    }

    @Override
    public void onActivityResumed(@NonNull Activity activity) {
    }

    @Override
    public void onActivityStopped(@NonNull Activity activity) {
    }

    @Override
    public void onActivitySaveInstanceState(@NonNull Activity activity, @Nullable Bundle bundle) {
    }

    @Override
    public void onActivityDestroyed(@NonNull Activity activity) {
    }

}

File: urbanairship-core/src/main/java/com/urbanairship/app/SimpleActivityListener.kt
/* Copyright Airship and Contributors */
package com.urbanairship.app

import android.app.Activity
import android.os.Bundle

/**
 * A convenience class to extend when you only want to listen for a subset
 * of of activity events.
 */
public open class SimpleActivityListener public constructor() : ActivityListener {

    override fun onActivityPaused(activity: Activity) { }

    override fun onActivityCreated(activity: Activity, bundle: Bundle?) { }

    override fun onActivityStarted(activity: Activity) { }

    override fun onActivityResumed(activity: Activity) { }

    override fun onActivityStopped(activity: Activity) { }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) { }

    override fun onActivityDestroyed(activity: Activity) { }
}
